backend/multi/asset.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multi
+
+import (
+	"fmt"
+
+	"perun.network/go-perun/channel"
+)
+
+type (
+	// Asset defines a multi-ledger asset.
+	Asset interface {
+		channel.Asset
+		LedgerID() LedgerID
+	}
+
+	// LedgerIDMapKey is the map key representation of a ledger identifier.
+	LedgerIDMapKey string
+
+	// LedgerID represents a ledger identifier.
+	LedgerID interface {
+		MapKey() LedgerIDMapKey
+	}
+
+	assets []channel.Asset
+)
+
+// LedgerIDs returns the identifiers of the associated ledgers.
+func (a assets) LedgerIDs() ([]LedgerID, error) {
```

Comment:
This is a helper function. No specific order is expected. The return value should be something that can be iterated on so I decided to use a slice because it seemed clean to me. If you insist, I can also return the map instead.